lib/purchases_flutter.dart

```diff
@@ -306,6 +306,33 @@ class Purchases {
     return customerInfo;
   }
 
+  /// Google Play only.
```

Comment:
Is this an okay way of doing this? Or should we also do a platform check in the code?

I mean... Getting a `SubscriptionOption` isn't even possible on iOS or Amazon so this can't really be called anyway 🤷‍♂️

Reply:
I would do whatever we do in `beginRefundRequestForActiveEntitlement`, which is throw an `UnsupportedPlatformException`. That's the most recent case where we've had to deal with one-platform functions and I think it's the most predictable and descriptive way.

Reply:
I think I'm confused on how and where this should be handle (especially since this is specifically for Google Android and not Amazon).

1) Is it best to do the check at the start of this method before invoking anything? If so. what is the best way to check if Amazon?
2) Or do we do this deeper into the stack? Like on an iOS and Android level? Do I need to add this method to iOS?

@tonidero @vegaro Not _as_ familiar here so any suggestion would be appreciated 😇

Reply:
I think the earlier the better. If we can do it in the flutter level, and that way avoid having to add the iOS methods, that makes the most sense.